app/views/Project/EntryEdit/LeftPane/AssistPopup/index.tsx

```diff
@@ -0,0 +1,351 @@
+                    widgetsHints.push({
+                        widgetPk: widget.id,
+                        widgetType: 'NUMBER',
+                        hints: mockAssistedMappingsResponse.numbers,
+                    });
+                }
+            }
+            if (
+                widget.widgetId === 'GEO'
+                && mockAssistedMappingsResponse.locations.length > 0
+            ) {
+                widgetsHints.push({
+                    widgetPk: widget.id,
+                    widgetType: 'GEO',
+                    hints: mockAssistedMappingsResponse.locations,
+                });
+            }
+            return undefined;
+        });
+
+        const recommentedAttributesMap = listToMap(
+            recommendedAttributes,
+            (attr) => attr.widget,
+            (attr) => attr,
+        );
+
+        setAllHints(widgetsHints.filter(isDefined));
+
+        onChange(
+            (oldEntry) => {
+                if (!oldEntry) {
+                    const newEntry = {
+                        clientId: randomString(),
+                        entryType: 'EXCERPT' as const,
+                        lead: leadId,
+                        excerpt: selectedText,
+                        droppedExcerpt: selectedText,
+                        attributes: recommendedAttributes,
+                    };
+
+                    return newEntry;
+                }
+                const oldAttributes = oldEntry?.attributes ?? [];
+
+                // NOTE: Updating the existing attributes
+                // FIXME: Currently overrides all info, maybe we should only update data
+                const updatedAttributes = oldAttributes.map((attr) => (
+                    recommentedAttributesMap[attr.widget] ? ({
+                        ...recommentedAttributesMap[attr.widget],
+                        clientId: attr.clientId,
+                        widget: attr.widget,
+                        id: attr.id,
+                        widgetVersion: attr.widgetVersion,
+                    }) : (
+                        attr
+                    )
+                ));
+
+                const oldAttributesMap = listToMap(
+                    oldAttributes,
+                    (attr) => (attr.widget),
+                    () => true,
+                );
+                const removedNewAttributes = recommendedAttributes.filter(
+                    (attr) => !oldAttributesMap[attr.widget],
+                );
+
+                // NOTE: Adding new attributes from suggestion and removing duplicates
+                const newAttributes = unique([
+                    ...removedNewAttributes,
+                    ...updatedAttributes,
+                ], (attr) => attr.widget);
+
+                // FIXME: Spreading newEntry does not seem right
+                // need to discuss
+                return {
+                    ...oldEntry,
+                    attributes: newAttributes,
+                };
```

Comment:
Let's create a generic function to update existing features, add new features and delete features that are omitted